README.md
![logo](https://user-images.githubusercontent.com/4670058/147123484-9f43b7a7-6ffb-4eaa-a991-bd9a716a0f4a.png)

[themarkets.io](https://themarkets.io/)

## Build Setup

```bash
# install dependencies
$ cd frontend
$ npm install
$ cd ..
$ cd backend
$ npm install

# run from root folder not `frontend` or 'backend'
$ cd ..
$ yarn develop

# build for production
$ npm run build

# deploy
$ cd frontend
$ npm run build
$ firebase deploy

# run firebase emulator for crypto cloud functions
$ cd frontend
$ npm run-script generate
$ firebase emulators:start
```

-------------

### Overview

Our app started off focussing on the traditional markets but we're now shifting focus to crypto, memestocks, defi, gamefi, nfts, carbon offsets and other emerging markets such as the metaverse. The markets have gone through a revolution in the last few years most exemplified by Gamestop. Now a meme can turn a failing company into a billion-dollar success. We want to provide a hub for new investors to find the latest news, trends and market prices.

---

### Data

Currently our main data provider is finage for stocks, crypto, indices, bonds, forex and news. We also use coinmarketcap to fetch the top 50 crypto coins by market cap since finage does not provide this data. Recently we added Trading Economics api data to collect commodities including EU Carbon Offsets.

We are trialling [Stockpulse](https://stockpulse.ai/) as a potential replacement to finage. We have a free trial which expires on January 31st 2022. Their data also includes market sentiment analysis which they call 'buzz'. This could be useful for further developing our app.

---

### Folder structure

From the root folder you will see the top folders 'frontend' and 'backend'. 'frontend' is for the main app while 'backend' is for Strapi. From the root folder you can run `yarn develop` to run both locally in dev mode.

Currently the lists of market categories are kept in market.js, this is used for fetching data and rendering IndexList components throughout the app.

The layouts/default file currently takes care of all data collection processes. Rather than use the store, data is emitted to pages and components via the event bus.

Within the pages directory we have subdirectories for each market category i.e. /cryptocurrency. Each subdirectory contains an index page which lists all symbols of it's type and a _symbol page for individual symbol details and charts. There are also subdirectories for our different Strapi content sections - personal-finance, analysis, magazine, trading-101.

---

### Components

I will just cover the most important, heavily-used components.

##### IndexList > Price

The IndexList render the lists of each category on the homepage and each category index page. They require different style depending on which page and category they are rendered. For index pages we use a grid style and for movers/rising lists the style is also slightly different. A sub component of the IndexList is Price, this is necessary for updating the price data and show visual changes, i.e. flashing green/red. The Price layout varies dependant on the list in which it has been rendered as well.

##### Item > Chart

The Item component is used on every _symbol page to render the main content of the page. This helps maintain consistentcy across the categories. Some pages also render the latest news on their symbol and info about the instrument such as market cap for a coin. The main feature of each symbol page is the Chart. Originally we used highcharts for this, however, we have recently switched to using TradingView since they provide Candlestick charts which update in real-time. When users select a short time range such as 1/5/15 minutes we fetch new aggregate data so the charts can update in real-time.

##### Header > Ticker > TickerItem

The Header contains the desktop and mobile navigation menus and the Ticker bar. The Ticker displays a small number of top symbols. Each symbol is rendered via the TickerItem component. The Ticker component is also used for styling the icons throughout the site. The icon class is set via the market.js file. It would be nice to set class names dynamically instead of hardcoding, however, we use PNG , SVGs and GIfs depending on the category so this is currently the best way of styling icons. Ideally the data provider would include icon data instead.

---

### Current issues

There have been many issues with our current data provider, finage, unstable connections, missing endpoints for certain symbols, the aggregate endpoint is inconsistent. We’ve been waiting months for finage to release CFD/Futures data for indices. We use websockets for live data streaming, the connection isn’t always stable. We also need to display when each symbols market is open, the finage market indicators api is quite basic, it just shows if an entire market is open, i.e. forex / nasdaq.

We need to refactor our data collection architecture, currently all api and websocket calls are made in layouts/default. Preferably the data is fetched by a standalone node.js app which then feeds this through to the main app, runs cron jobs on set intervals to keep lists up to date such as the top 50 crypto coins and takes CPU load off the main app.

---

### Backlog

The full backlog of issues and requested features can be found [here](https://urala.atlassian.net/browse/UA-7)

---

### Roadmap

#### Completed milestones
- App research and development began December 2020
- Data provider secured February 2021
- MVP design finished April 2021
- MVP launched October 2021
- Strapi CMS added November 2021
- Candlestick charts added December 2021
- Search function for all symbols with autosuggest added January 2022

#### Future milestones
- Add indice CFDs and Futures
- Refactor data collection via node.js app and GraphQL
- Toggle index list to cards
- Add crypto coin categories DeFi, GameFi, DAO, Memecoins, NFTs
- Redesign UI with a gamified, pixel art style
- User accounts
- Social features: comments threads, discord, SNS
- Integrate lightstreamer

---

#### Strapi CMS

We are using Strapi CMS for our content section, currently this is just /personal-finance articles. However, we intend to continue growing this area of the site.

--------------

### Cryptocurrency
Top 50 by marketcap

### Forex
EURUSD, USDJPY, USDKRW, USDTRY, GBPUSD, USDBRL, USDILS, USDRUB

### Commodities
EU Carbon Offsets, Gold, Silver, Natural Gas, Crude Oil, Brent Oil, Coffee, Platinum, Cobalt, Solar Energy Index, Wind Energy Index, CRB Index, LME Index

### Stocks
Apple, Pfizer, Amazon, Facebook, Microsoft, Moderna, NIO, Tesla, Nvidia, Xpeng, Palantir, Boeing, Santander, Alibaba, Gamestop, AMC, BlackBerry

### Indices
NASDAQ, NASDAQ CFD, NASDAQ Futures, DOW JONES, DOW JONES CFD, DOW JONES Futures, S&P 500, S&P 500 CFD, S&P 500 Futures, FTSE, FTSE CFD, FTSE Futures, NIKKEI, NIKKEI CFD, NIKKEI Futures, HANG SENG, HANG SENG CFD, HANG SENG Futures, SHANGHAI, SHANGHAI CFD, SHANGHAI Futures, KOSPI, KOSPI CFD, KOSPI Futures, DAX, DAX CFD, DAX Futures, IBEX 35, IBEX 35 CFD, IBEX Futures, BIST 100, BIST 100 CFD, BIST 100 Futures, Dollar Index

### Bonds
US 1M Bond, US 3M Bond, US 6M Bond, US 1 Year Bond, US 2 Year Bond, US 5 Year Bond, US 7 Year Bond, US 10 Year Bond, US 20 Year Bond, US 30 Year Bond



frontend/CHANGELOG
Fix news displaying bug on commodities index
Index using example api calls

frontend/market.js
export let cryptocurrency = [
  {name: 'Bitcoin', symbol: 'BTCUSD', live: 'BTCUSDT', icon: 'btc', thumb: 'bitcoin.jpg'},
  {name: 'Ethereum', symbol: 'ETHUSD', live: 'ETHUSDT', icon: 'eth', thumb: 'ethereum.png'},
  {name: 'BNB', symbol: 'BNBUSD', live: 'BNBUSDT', icon: 'bnb', thumb: 'bnb.png'},
  {name: 'Cardano', symbol: 'ADAUSD', live: 'ADAUSDT', icon: 'ada', thumb: 'cardano.jpg'},
  {name: 'Solana', symbol: 'SOLUSD', live: 'SOLUSDT', icon: 'sol', thumb: 'sol.png'},
  {name: 'XRP', symbol: 'XRPUSD', live: 'XRPUSDT', icon: 'xrp', thumb: 'xrp.png'},
  {name: 'Polkadot', symbol: 'DOTUSD', live: 'DOTUSDT', icon: 'dot', thumb: 'polkadot.png'},
  {name: 'Dogecoin', symbol: 'DOGEUSD', live: 'DOGEUSDT', icon: 'doge', thumb: 'doge.jpg'},
  {name: 'Shiba Inu', symbol: 'SHIBUSD', live: 'SHIBUSDT', icon: 'shib', thumb: 'shiba.png'},
  {name: 'Terra', symbol: 'LUNAUSD', live: 'LUNAUSDT', icon: 'luna', thumb: 'luna.png'},
  {name: 'Avalanche', symbol: 'AVAXUSD', live: 'AVAXUSDT', icon: 'avax', thumb: 'avax.png'},
  {name: 'Litecoin', symbol: 'LTCUSD', live: 'LTCUSDT', icon: 'ltc', thumb: 'ltc.png'},
  {name: 'Uniswap', symbol: 'UNIUSD', live: 'UNIUSDT', icon: 'uni', thumb: 'uni.png'},
  {name: 'Chainlink', symbol: 'LINKUSD', live: 'LINKUSDT', icon: 'link', thumb: 'link.png'},
  {name: 'Polygon', symbol: 'MATICUSD', live: 'MATICUSDT', icon: 'matic', thumb: 'matic.png'},
  {name: 'Algorand', symbol: 'ALGOUSD', live: 'ALGOUSDT', icon: 'algo', thumb: 'algo.png'},
  {name: 'VeChain', symbol: 'VETUSD', live: 'VETUSDT', icon: 'vet', thumb: 'vet.png'},
  {name: 'Stellar', symbol: 'XLMUSD', live: 'XLMUSDT', icon: 'xlm', thumb: 'xlm.png'},
  {name: 'Axie Infinity', symbol: 'AXSUSD', live: 'AXSUSDT', icon: 'axs', thumb: 'axie.png'},
  {name: 'Filecoin', symbol: 'FILUSD', live: 'FILUSDT', icon: 'fil', thumb: 'file.png'},
  {name: 'TRON', symbol: 'TRXUSD', live: 'TRXUSDT', icon: 'trx', thumb: 'tron.png'},
  {name: 'Ethereum Classic', symbol: 'ETCUSD', live: 'ETCUSDT', icon: 'etc', thumb: 'etc.png'},
  {name: 'Cosmos', symbol: 'ATOMUSD', live: 'ATOMUSDT', icon: 'atom', thumb: 'atom.png'},
  {name: 'THETsansA', symbol: 'THETAUSD', live: 'HETAUSDT', icon: 'theta', thumb: 'theta.png'},
  {name: 'FTX Token', symbol: 'FTTUSD', live: 'FTTUSDT', icon: 'ftt', thumb: 'ftt.png'},
  {name: 'Fantom', symbol: 'FTMUSD', live: 'FTMUSDT', icon: 'ftm', thumb: 'ftm.png'},
  {name: 'Hedera', symbol: 'HBARUSD', live: 'HBARUSDT', icon: 'hbar', thumb: 'hbar.png'},
  {name: 'Dai', symbol: 'DAIUSD', live: 'DAIUSDT', icon: 'dai', thumb: 'dai.png'},
  {name: 'Elrond', symbol: 'EGLDUSD', live: 'EGLDUSDT', icon: 'egld', thumb: 'egld.png'},
  {name: 'NEAR Protocol', symbol: 'NEARUSD', live: 'NEARUSDT', icon: 'near', thumb: 'near.png'},
  {name: 'The Graph', symbol: 'GRTUSD', live: 'GRTUSDT', icon: 'grt', thumb: 'grt.png'},
  {name: 'Tezos', symbol: 'XTZUSD', live: 'XTZUSDT', icon: 'xtz', thumb: 'xtz.png'},
  {name: 'Monero', symbol: 'XMRUSD', live: 'XMRUSDT', icon: 'xmr', thumb: 'xmr.png'},
  {name: 'EOS', symbol: 'EOSUSD', live: 'EOSUSDT', icon: 'eos', thumb: 'eos.png'},
  {name: 'Decentraland', symbol: 'MANAUSD', live: 'MANAUSDT', icon: 'mana', thumb: 'mana.png'},
  {name: 'Helium', symbol: 'HNTUSD', live: 'HNTUSDT', icon: 'hnt', thumb: 'hnt.png'},
  {name: 'PancakeSwap', symbol: 'CAKEUSD', live: 'CAKEUSDT', icon: 'cake', thumb: 'cake.png'},
  {name: 'Aave', symbol: 'AAVEUSD', live: 'AAVEUSDT', icon: 'aave', thumb: 'aave.png'},
  {name: 'Loopring', symbol: 'LRCUSD', live: 'LRCUSDT', icon: 'lrc', thumb: 'lrc.png'},
  {name: 'IOTA', symbol: 'IOTAUSD', live: 'IOTAUSDT', icon: 'miota', thumb: 'iota.png'},
  {name: 'Neo', symbol: 'NEOUSD', live: 'NEOUSDT', icon: 'neo', thumb: 'neo.png'},
  {name: 'Maker', symbol: 'MKRUSD', live: 'MKRUSDT', icon: 'mkr', thumb: 'mkr.png'},
  {name: 'Enjin', symbol: 'ENJUSD', live: 'ENJUSDT', icon: 'enj', thumb: 'enj.png'},
  {name: 'Dash', symbol: 'DASHUSD', live: 'DASHUSDT', icon: 'dash', thumb: 'dash.png'},
  {name: 'Compound', symbol: 'COMPUSD', live: 'COMPUSDT', icon: 'comp', thumb: 'comp.png'},
  {name: 'CRV', symbol: 'CRVUSD', live: 'CRVUSDT', icon: 'crv', thumb: 'crv.png'},
  {name: 'BAT', symbol: 'BATUSD', live: 'BATUSDT', icon: 'bat', thumb: 'bat.png'},
  {name: 'SushiSwap', symbol: 'SUSHIUSD', live: 'SUSHIUSDT', icon: 'sushi', thumb: 'sushi.png'},
  {name: 'Zilliqa', symbol: 'ZILUSD', live: 'ZILUSDT', icon: 'zil', thumb: 'zil.png'},
  {name: 'yearn.finance', symbol: 'YFIUSD', live: 'YFIUSDT', icon: 'yfi', thumb: 'yfi.png'},
  {name: '1inch', symbol: '1INCHUSD', live: '1INCHUSDT', icon: 'oneinch', thumb: 'oneinch.png'},
];
export let defi = [
  {name: 'Curve', symbol: 'CRV'},
  {name: 'Terra', symbol: 'LUNA'},
  {name: 'Chainlink', symbol: 'LINK'},
  {name: 'Uniswap', symbol: 'UNI'},
  {name: 'Dai', symbol: 'DAI'},
  {name: 'cETH', symbol: 'CETH', icon: 'ceth'},
  {name: 'Lido Staked Ether', symbol: 'STETH', icon: 'lido'},
  {name: 'The Graph', symbol: 'GRT'},
  {name: 'cDAI', symbol: 'CDAI', icon: 'cdai'},
  {name: 'PancakeSwap', symbol: 'CAKE'},
  {name: 'Aave', symbol: 'AAVE'},
];
// NFTs
// Axie Infinity AXS
// Theta Network THETA
// Decentraland MANA
// Flow FLOW
// Chiliz CHZ
// Enjin Coin ENJ
// The Sandbox SAND
// ECOMI OMI
// WAX WAXP
// Audius AUDIO
export let currencies = [
  {name: 'EUR/USD', symbol: 'EURUSD', icon: 'eurusd', type: 'currency'},
  {name: 'USD/JPY', symbol: 'USDJPY', icon: 'usdjpy', type: 'currency'},
  {name: 'USD/KRW', symbol: 'USDKRW', icon: 'usdkrw', type: 'currency'},
  {name: 'USD/TRY', symbol: 'USDTRY', icon: 'usdtry', type: 'currency'},
  {name: 'GBP/USD', symbol: 'GBPUSD', icon: 'gbpusd', type: 'currency'},
  {name: 'USD/BRL', symbol: 'USDBRL', icon: 'usdbrl', type: 'currency'},
  {name: 'USD/ILS', symbol: 'USDILS', icon: 'usdils', type: 'currency'},
  {name: 'USD/RUB', symbol: 'USDRUB', icon: 'usdrub', type: 'currency'},
  {name: 'Dollar Index', symbol: 'DXY', icon: 'usa', type: 'currency'},
  {name: 'XAU/USD', symbol: 'XAUUSD', icon: 'gold', type: 'commodity'},
  {name: 'XAG/USD', symbol: 'XAGUSD', icon: 'silver', type: 'commodity'},
  {name: 'WTI/USD', symbol: 'WTIUSD', icon: 'crude-oil', type: 'commodity'},
  {name: 'XBR/USD', symbol: 'XBRUSD', icon: 'brent-oil', type: 'commodity'},
];
export let stocks = [
  {name: 'Apple', symbol: 'AAPL', icon: 'apple', thumb: 'apple.jpg'},
  {name: 'Pfizer', symbol: 'PFE', icon: 'pfizer', thumb: 'pfizer.jpg'},
  {name: 'Amazon', symbol: 'AMZN', icon: 'amazon', thumb: 'amazon.jpg'},
  {name: 'Facebook', symbol: 'FB', icon: 'facebook', thumb: 'facebook.jpg'},
  {name: 'Microsoft', symbol: 'MSFT', icon: 'microsoft', thumb: 'microsoft.png'},
  {name: 'Moderna', symbol: 'MRNA', icon: 'moderna', thumb: 'moderna.jpg'},
  {name: 'NIO', symbol: 'NIO', icon: 'nio', thumb: 'nio.jpg'},
  {name: 'Tesla', symbol: 'TSLA', icon: 'tesla', thumb: 'tesla.jpg'},
  {name: 'Nvidia', symbol: 'NVDA', icon: 'nvidia', thumb: 'nvidia.png'},
  {name: 'Xpeng', symbol: 'XPEV', icon: 'xpeng', thumb: 'xpeng.png'},
  {name: 'Palantir', symbol: 'PLTR', icon: 'palantir', thumb: 'palantir.png'},
  {name: 'Boeing', symbol: 'BA', icon: 'boeing', thumb: 'boeing.jpg'},
  {name: 'Santander', symbol: 'SAN', icon: 'santander', thumb: 'santander.png'},
  {name: 'Alibaba', symbol: 'BABA', icon: 'alibaba', thumb: 'alibaba.png'},
  {name: 'Gamestop', symbol: 'GME', icon: 'gme', thumb: 'gamestop.png'},
  {name: 'AMC', symbol: 'AMC', icon: 'amc', thumb: 'amc.jpg'},
  {name: 'BlackBerry', symbol: 'BB', icon: 'blackberry', thumb: 'blackberry.png'}
];
export let indices = [
  {name: 'NASDAQ', symbol: 'IXIC', icon: 'usa', type: 'indice'},
  {name: 'NASDAQ RTD*', symbol: 'IXIC', icon: 'usa', type: 'indice', cfd: true, country: "US"},
  {name: 'DOW JONES', symbol: 'DJI', icon: 'usa', type: 'indice'},
  {name: 'DOW JONES RTD*', symbol: 'DJI', icon: 'usa', type: 'indice', cfd: true, country: "US"},
  {name: 'S&P 500', symbol: 'GSPC', icon: 'usa', type: 'indice'},
  {name: 'S&P 500 RTD*', symbol: 'GSPC', icon: 'usa', type: 'indice', cfd: true, country: "US"},
  {name: 'FTSE', symbol: 'FTSE', icon: 'uk', type: 'indice'},
  {name: 'FTSE RTD*', symbol: 'FTSE', icon: 'uk', type: 'indice', cfd: true, country: "GB"},
  {name: 'NIKKEI', symbol: 'N225', icon: 'jp', type: 'indice'},
  {name: 'NIKKEI Futures RTD*', symbol: 'N225', icon: 'jp', type: 'indice', cfd: true, country: "JP"},
  {name: 'HANG SENG', symbol: 'HSI', icon: 'hk', type: 'indice'},
  {name: 'HANG SENG RTD*', symbol: 'HSI', icon: 'hk', type: 'indice', cfd: true, country: "HK"},
  {name: 'SHANGHAI', symbol: '000001', icon: 'cn', type: 'indice'},
  {name: 'SHANGHAI RTD*', symbol: '000001', icon: 'cn', type: 'indice', cfd: true, country: "CN"},
  {name: 'KOSPI', symbol: 'KS11', icon: 'ko', type: 'indice'},
  {name: 'KOSPI RTD*', symbol: 'KS11', icon: 'ko', type: 'indice', cfd: true, country: "KO"},
  {name: 'DAX', symbol: 'GDAXI', icon: 'germany', type: 'indice'},
  {name: 'DAX RTD*', symbol: 'GDAXI', icon: 'germany', type: 'indice', cfd: true, country: "DE"},
  {name: 'IBEX 35', symbol: 'IBEX', icon: 'spain', type: 'indice'},
  {name: 'IBEX 35 RTD*', symbol: 'IBEX', icon: 'spain', type: 'indice', cfd: true, country: "ES"},
  {name: 'BIST 100', symbol: 'XU100', icon: 'turkey', type: 'indice'},
  {name: 'BIST 100 RTD*', symbol: 'XU100', icon: 'turkey', type: 'indice', cfd: true, country: "TR"},
  {name: 'Dollar Index', symbol: 'DXY', icon: 'usa', type: 'currency'},
  {name: 'US 1M Bond', symbol: 'DGS1MO', icon: 'usa'},
  {name: 'US 3M Bond', symbol: 'DGS3MO', icon: 'usa'},
  {name: 'US 6M Bond', symbol: 'DGS6MO', icon: 'usa'},
  {name: 'US 1 Year Bond', symbol: 'DGS1', icon: 'usa'},
  {name: 'US 2 Year Bond', symbol: 'DGS2', icon: 'usa'},
  {name: 'US 5 Year Bond', symbol: 'DGS5', icon: 'usa'},
  {name: 'US 7 Year Bond', symbol: 'DGS7', icon: 'usa'},
  {name: 'US 10 Year Bond', symbol: 'DGS10', icon: 'usa'},
  {name: 'US 20 Year Bond', symbol: 'DGS20', icon: 'usa'},
  {name: 'US 30 Year Bond', symbol: 'DGS30', icon: 'usa'},
];
export let bonds = [
  {name: 'US 1M Bond', symbol: 'DGS1MO', icon: 'usa'},
  {name: 'US 3M Bond', symbol: 'DGS3MO', icon: 'usa'},
  {name: 'US 6M Bond', symbol: 'DGS6MO', icon: 'usa'},
  {name: 'US 1 Year Bond', symbol: 'DGS1', icon: 'usa'},
  {name: 'US 2 Year Bond', symbol: 'DGS2', icon: 'usa', type: 'homeBond'},
  {name: 'US 5 Year Bond', symbol: 'DGS5', icon: 'usa', type: 'homeBond'},
  {name: 'US 7 Year Bond', symbol: 'DGS7', icon: 'usa'},
  {name: 'US 10 Year Bond', symbol: 'DGS10', icon: 'usa', type: 'homeBond'},
  {name: 'US 20 Year Bond', symbol: 'DGS20', icon: 'usa', type: 'homeBond'},
  {name: 'US 30 Year Bond', symbol: 'DGS30', icon: 'usa', type: 'homeBond'},
];

export let rising = []
// export let commodities = [
//   // {name: 'Carbon Emissions', symbol: 'CARBON', icon: 'carbon'},
//   {name: 'Gold', symbol: 'XAUUSD', icon: 'gold'},
//   {name: 'Silver', symbol: 'XAGUSD', icon: 'silver'},
//   // {name: 'Bitcoin Futures', symbol: 'BTCUSD', icon: 'btc'},
//   // {name: 'Natural Gas', symbol: 'NGAS', icon: 'natgas'},
//   {name: 'Crude Oil', symbol: 'WTIUSD', icon: 'crude-oil'},
//   {name: 'Brent Oil', symbol: 'XBRUSD', icon: 'brent-oil'},
//   // {name: 'US Coffee C', symbol: 'USCOFFEEC', icon: 'coffee'},
// ];

export let commodities = [
  {name: 'EU Carbon Emissions', symbol: 'EECXM:IND', icon: 'carbon'},
  {name: 'Gold', symbol: 'XAUUSD:CUR', icon: 'gold'},
  {name: 'Silver', symbol: 'XAGUSD:CUR', icon: 'silver'},
  {name: 'Natural Gas', symbol: 'NG1:COM', icon: 'natgas'},
  {name: 'Crude Oil', symbol: 'CL1:COM', icon: 'crude-oil'},
  {name: 'Brent Oil', symbol: 'CO1:COM', icon: 'brent-oil'},
  {name: 'Coffee', symbol: 'KC1:COM', icon: 'coffee'},
  {name: 'Platinum', symbol: 'XPTUSD:CUR', icon: 'platinum'},
  {name: 'Cobalt', symbol: 'LCO1:COM', icon: 'cobalt'},
  {name: 'Solar Energy Index', symbol: 'SOLARNTR:IND', icon: 'solar'},
  {name: 'Wind Energy Index', symbol: 'GWETR:IND', icon: 'wind'},
  {name: 'CRB Index', symbol: 'CRYTR:IND', icon: 'crb'},
  {name: 'LME Index', symbol: 'LME:IND', icon: 'lme'},
];